cmd/jaeger/internal/extension/jaegerstorage/converter/reader.go

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package converter
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+
+	jaeger2otlp "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/jaeger"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+
+	"github.com/jaegertracing/jaeger/model"
+	spanstore_v1 "github.com/jaegertracing/jaeger/storage/spanstore"
+	"github.com/jaegertracing/jaeger/storage_v2/spanstore"
+)
+
+type TraceReader struct {
+	spanReader spanstore_v1.Reader
+}
+
+func NewTraceReader(spanReader spanstore_v1.Reader) (spanstore.Reader, error) {
+	return &TraceReader{
+		spanReader: spanReader,
+	}, nil
+}
+
+// GetTrace implements spanstore.Reader.
+func (s *TraceReader) GetTrace(ctx context.Context, traceID pcommon.TraceID) (ptrace.Traces, error) {
+	// otelcol-contrib has the translator to jaeger proto but declared in private function
+	// similar to https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/internal/coreinternal/idutils/big_endian_converter.go#L21
+	traceIDHigh, traceIDLow := binary.BigEndian.Uint64(traceID[:8]), binary.BigEndian.Uint64(traceID[8:])
+	id := model.TraceID{
```

Comment:
I think we may have some marshalling code in `model` to convert by array to TraceID